biz.aQute.bndlib.tests/test/test/ProcessorTest.java

```diff
@@ -850,4 +855,73 @@ public void testProvenance() throws IOException {
 
 	}
 
+	@Test
+	public void testWarningOnCounterIntuitiveInclude(@InjectTemporaryDirectory
+	File base, SoftAssertions softly) throws IOException {
+		try {
+			final File bnd = new File(base, "bnd.bnd");
+			IO.store("""
+				-include a.bnd
+				Header1: i will lose
+				""", bnd);
+			File sup = new File(base, "a.bnd");
+			IO.store("""
+				Header1: i will win
+				""", sup);
+
+			try (Processor a = new Processor()) {
+				a.setProperties(bnd);
+
+				softly.assertThat(a.getProperty("Header1"))
+					.isEqualTo("i will win");
+				softly.assertThat(a.getWarnings())
+					.containsExactly(
+						"[Include Override]: `Header1` declaration is overridden by -include: a.bnd and thus ignored (consider using -include: ~a.bnd).");
+
+
+			}
+
+			// now prevent the overwrite by doing what the warning says, and use
+			// the '~' (tilde)
+
+			IO.store("""
+				-include: ~a.bnd
+				Header1: now i will win, and a.bnd will not overwrite
+				""", bnd);
+
+			try (Processor a = new Processor()) {
+				a.setProperties(bnd);
+
+				softly.assertThat(a.getProperty("Header1"))
+					.isEqualTo("now i will win, and a.bnd will not overwrite");
+				softly.assertThat(a.getWarnings())
+					.isEmpty();
+
+			}
+
+			// now test a case where an overwrite will happen,
+			// but no warning is logged, because the overwrite will overwrite
+			// with the same value
+
+			IO.store("""
+				-include: a.bnd
+				Header1: i will win
+				""", bnd);
+
+			try (Processor a = new Processor()) {
+				a.setProperties(bnd);
+
+				softly.assertThat(a.getProperty("Header1"))
+					.isEqualTo("i will win");
+				softly.assertThat(a.getWarnings())
+					.isEmpty();
+
+			}
+
+		} finally {
+			IO.delete(base);
```

Comment:
This finally block is unnecessary as the temp dir extension will clean it up by default.

If you ever need to disable this for debugging purposes, set `clear=false` on the annotation.

Reply:
good catch.